tests/test_rpm_verifier.py

```diff
@@ -381,5 +356,30 @@ def test_main(
         )
 
         assert mock_image_processor.return_value.call_count == 1
+        generate_output_mock.assert_called_once_with([sentinel.output])
         mock_image_processor.return_value.assert_has_calls([call("img1")])
-        mock_generate_output.assert_called_once_with([sentinel.output], True)
+
+    def test_main_fail_on_unsigned_rpm_or_errors(
+        self,
+        create_generate_output_mock: MagicMock,
+        mock_image_processor: MagicMock,
+    ) -> None:
+        """Test call to rpm_verifier.py main function fails
+        when whe 'fail-unsigned' flag is used and there are unsigned RPMs
+        """
+        fail_unsigned: bool = True
+        create_generate_output_mock(with_failures=True)
+        with pytest.raises(SystemExit):
+            rpm_verifier.main(  # pylint: disable=no-value-for-parameter
+                args=[
+                    "--input",
+                    "img1",
+                    "--fail-unsigned",
+                    fail_unsigned,
+                    "--workdir",
+                    "some/path",
+                ],
+                obj={},
+                standalone_mode=False,
+            )
+        assert mock_image_processor.return_value.call_count == 1
```

Comment:
We can actually assert on the error being printed, rather than on the call_count to image_processor which is not very related to the test.

```
        def _mock_generate_output(with_failures: bool = False) -> MagicMock:
            """Monkey-patched generate_output"""
            mock = create_autospec(generate_output, return_value=(with_failures, sentinel.output_gen_out))
...
        with pytest.raises(SystemExit) as err:
            ...
        assert err.value.code == sentinel.output_gen_out
```

Reply:
Fair point, fixed